src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -871,6 +871,18 @@ public AsyncIndex getAsyncIndexConnection(String indexName) throws PineconeValid
         return new AsyncIndex(connection, indexName);
     }
 
+    /**
+     * A method to create and return a new instance of the {@link Inference} client.
+     * <p>
+     * This method initializes the Inference client using the current ApiClient
+     * from the {@link ManageIndexesApi}. The Inference client can then be used
+     * to interact with Pinecone's embedding API.
+     * @return A new {@link Inference} client instance.
+     */
```

Comment:
We should include guidance about whether this instance should be cached and re-used, and whether it's thread-safe

Reply:
Added the instructions in the README and in the Pinecone class.